File: Java/StrategyPattern/src/com/laxmena/VideoChunk.java
package com.laxmena;

public class VideoChunk {
    private int chunkId;
    private String videoChunkContent;

    public VideoChunk(int chunkId, String content) {
        this.chunkId = chunkId;
        this.videoChunkContent = content;
    }

    public VideoChunk(int chunkId) {
        this.chunkId = chunkId;
    }

    // Getter methods
    public int getChunkId() {
        return chunkId;
    }

    public String getVideoChunkContent() {
        return videoChunkContent;
    }

    public int getVideoChunkSize() {
        return videoChunkContent.length();
    }

    // Setter methods
    public void setVideoChunkContent(String videoChunkContent) {
        this.videoChunkContent = videoChunkContent;
    }
}

File: Strategy/Python/VideoStreamingService/VideoChunk/video_chunk.py
class VideoChunk:
    """
    VideoChunk is representation of a Video Chunk Object.
    In this example, we do not download the data from Server, instead we simulate the VideoChunk data.
    So Constructor methods are defined to set the contents of the VideoChunks.
    VideoChunk instances are created in the concrete strategy implementations.
    """
    def __init__(self, chunk_id):
        self.chunk_id = chunk_id
        self.content = None

    def __init__(self, chunk_id, content):
        self.chunk_id = chunk_id
        self.content = content

    # getter method
    def get_chunk_id(self):
        return self.chunk_id

    # getter method
    def get_content(self):
        return self.content

    # setter method
    def set_content(self, content):
        self.content = content